lib/pacemaker/pcmk_verify.c

```diff
@@ -130,32 +128,24 @@ pcmk_verify(xmlNodePtr *xml, const char *cib_source)
     pcmk__register_lib_messages(out);
 
     rc = pcmk__parse_cib(out, cib_source, &cib_object);
-    
     if (rc != pcmk_rc_ok) {
         out->err(out, "Couldn't parse input");
         goto done;
     }
 
     scheduler = pe_new_working_set();
-
     if (scheduler == NULL) {
         rc = errno;
-        out->err(out, "Could not allocate scheduler data: %s", pcmk_rc_str(rc));
+        out->err(out, "Couldn't allocate scheduler data: %s", pcmk_rc_str(rc));
         goto done;
     }
-    
-    scheduler->priv = out;
 
+    scheduler->priv = out;
     rc = pcmk__verify(scheduler, out, cib_object);
-    
-    pe_free_working_set(scheduler);
 
 done:
     pcmk__xml_output_finish(out, pcmk_rc2exitc(rc), xml);
-
-    if (cib_object != NULL) {
-        free_xml(cib_object);      
-    }
-
+    free_xml(cib_object);
+    pe_free_working_set(scheduler);
```

Comment:
I'll move this above `pcmk__xml_output_finish()` since it contains a pointer to `out`